pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1456,3 +1456,87 @@ func TestShowLogicalReplicationJobs(t *testing.T) {
 	jobutils.WaitForJobToCancel(t, dbA, jobAID)
 	jobutils.WaitForJobToCancel(t, dbA, jobBID)
 }
+
+// TestUserPrivileges verifies the grants and role permissions
+// needed to start and administer LDR
+func TestUserPrivileges(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	clusterArgs := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestControlsTenantsExplicitly,
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			},
+		},
+	}
+
+	server, s, dbA, _ := setupLogicalTestServer(t, ctx, clusterArgs, 1)
+	defer server.Stopper().Stop(ctx)
+
+	dbBURL, cleanupB := s.PGUrl(t, serverutils.DBName("b"))
+	defer cleanupB()
+
+	var (
+		jobAID jobspb.JobID
+	)
+
+	dbA.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab", dbBURL.String()).Scan(&jobAID)
+
+	// Create user with no privileges
+	dbA.Exec(t, fmt.Sprintf("CREATE USER %s", username.TestUser))
+	testuser := sqlutils.MakeSQLRunner(s.SQLConn(t, serverutils.User(username.TestUser), serverutils.DBName("a")))
+
+	t.Run("Viewjob test", func(t *testing.T) {
+		// NEED VIEWJOB system grant to view admin LDR jobs
+		result := testuser.QueryStr(t, "select * from [SHOW JOBS] where job_id=$1", jobAID)
+		require.Empty(t, result, "The user should see no rows without the VIEWJOB grant")
+
+		result = testuser.QueryStr(t, "select * from [SHOW LOGICAL REPLICATION JOBS] where job_id=$1", jobAID)
+		require.Empty(t, result, "The user should see no rows without the VIEWJOB grant")
+
+		var returnedJobID jobspb.JobID
+		dbA.Exec(t, fmt.Sprintf("GRANT SYSTEM VIEWJOB to %s", username.TestUser))
+		testuser.QueryRow(t, "select job_id from [SHOW JOBS] where job_id=$1", jobAID).Scan(&returnedJobID)
+		require.Equal(t, returnedJobID, jobAID, "The user should see the LDR job with the VIEWJOB grant")
+
+		testuser.QueryRow(t, "select job_id from [SHOW LOGICAL REPLICATION JOBS] where job_id=$1", jobAID).Scan(&returnedJobID)
+		require.Equal(t, returnedJobID, jobAID, "The user should see the LDR job with the VIEWJOB grant")
+	})
+
+	// Kill replication job so we can create one with the testuser
+	dbA.Exec(t, "CANCEL JOB $1", jobAID)
+	jobutils.WaitForJobToCancel(t, dbA, jobAID)
+
+	t.Run("Schema test", func(t *testing.T) {
+		dbA.Exec(t, "CREATE SCHEMA testschema")
+
+		// Ensure the testuser has CREATE privileges
+		testuser.ExpectErr(t, "user testuser does not have CREATE privilege on schema testschema", fmt.Sprintf(testingUDFAcceptProposedBaseWithSchema, "testschema", "tab"))
+		dbA.Exec(t, "GRANT CREATE ON SCHEMA testschema TO testuser")
+		testuser.Exec(t, fmt.Sprintf(testingUDFAcceptProposedBaseWithSchema, "testschema", "tab"))
+	})
+
+	t.Run("Replication grant test", func(t *testing.T) {
```

Comment:
s/Replication grant test/replication/r

notice that im proposing that you remove "test" from the name, as it is redundant, and that sub test names should be consistent.

Reply:
Sounds good, will adhere to this